src/promptflow/tests/sdk_cli_azure_test/_recording_base.py

```diff
@@ -0,0 +1,184 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+import inspect
+import unittest
+from pathlib import Path
+from typing import List, Type
+from urllib.parse import parse_qs, urlparse
+
+import vcr
+from vcr.request import Request
+
+from ._recording_processors import (
+    AzureWorkspaceTriadProcessor,
+    ConnectionProcessor,
+    DatastoreProcessor,
+    RecordingProcessor,
+    RequestUrlProcessor,
+    UploadHashProcessor,
+)
+from ._recording_utils import is_live, is_live_and_not_recording, sanitize_azure_workspace_triad
+
+
+class PFAzureIntegrationTestCase(unittest.TestCase):
+    FILTER_HEADERS = [
+        "authorization",
+        "client-request-id",
+        "retry-after",
+        "x-ms-client-request-id",
+        "x-ms-correlation-request-id",
+        "x-ms-ratelimit-remaining-subscription-reads",
+        "x-ms-request-id",
+        "x-ms-routing-request-id",
+        "x-ms-gateway-service-instanceid",
+        "x-ms-ratelimit-remaining-tenant-reads",
+        "x-ms-served-by",
+        "x-ms-authorization-auxiliary",
+        "x-ms-correlation-request-id",
+        "x-ms-ratelimit-remaining-subscription-writes",
+        "x-ms-request-id",
+        "x-ms-response-type",
+        "x-ms-routing-request-id",
+        "x-request-time",
+        "x-aml-cluster",
+        "aml-user-token",
+        "request-context",
+    ]
+
+    def __init__(self, method_name: str) -> None:
+        super(PFAzureIntegrationTestCase, self).__init__(method_name)
+
+        self.recording_processors = self._get_recording_processors()
+        self.playback_processors = self._get_playback_processors()
+
+        test_file_path = Path(inspect.getfile(self.__class__)).resolve()
+        recording_dir = (test_file_path.parent / "recordings").resolve()
+
+        self.is_live = is_live()
+
+        self.vcr = vcr.VCR(
+            cassette_library_dir=recording_dir.as_posix(),
+            before_record_request=self._process_request_recording,
+            before_record_response=self._process_response_recording,
+            decode_compressed_response=True,
+            record_mode="none" if not self.is_live else "all",
+            filter_headers=self.FILTER_HEADERS,
+        )
+        self.vcr.register_matcher("query", self._custom_request_query_matcher)
+
+        test_file_name = test_file_path.stem
+        test_file_recording_dir = (recording_dir / test_file_name).resolve()
+        test_file_recording_dir.mkdir(parents=True, exist_ok=True)
+        test_class_name = self.__class__.__name__
+        recording_filename = f"{test_class_name}_{method_name}.yaml"
+        self.recording_file = (test_file_recording_dir / recording_filename).resolve()
+
+        if self.is_live and not is_live_and_not_recording() and self.recording_file.is_file():
+            self.recording_file.unlink()
+
+    def setUp(self) -> None:
```

Comment:
why camel here

Reply:
unit test class method; have removed that for pure pytest experience